### ISSUE TITLE
Question on result of pretrain 433h and finetune 30h on LRS3

### ISSUE THREAD

**Issue by li563042811** [facebookresearch/av_hubert#66]
Hi,
    Recently I finished pretrain and finetune on LRS3 dataset which is pretraining on 433h and fine-tuning on 30h. 
    The pretrain config files I used is avhubert/conf/pretrain/base_lrs3_iter[1-5].yaml
    The fine-tuning config file I used is avhubert/conf/av-finetune/base_noise_pt_noise_ft_30h.yaml which deleted the parameters for noise.
    The decode result for 5 iterations is here
![image](https://user-images.githubusercontent.com/22743415/192981217-ea84acd7-a53c-47e1-809d-7af1c00f3a60.png)
    The result in your paper is here
![image](https://user-images.githubusercontent.com/22743415/192981640-dedc6d2d-56c4-47fe-aae2-0802ae02586f.png)

    So the audio-HuBERT on the same dataset is 4.9 in your paper. The AV-HuBERT‘s result is 6.73. I think there is something wrong in the training process, otherwise the AV-HuBERT‘s result should be better than audio-HuBERT's result which is 4.9.
    Here is my question about pretrain config file,
    1. modality_dropout is 0 in config file(iter 1-4) but in the paper it should be 0.5
    2. mask_prob_image is 0.8 but in the paper it should be 0.08
   
![image](https://user-images.githubusercontent.com/22743415/192984685-35d699d5-aeb8-4731-a549-c2d7129e6c6e.png)

   In a word, what should I do to reproduce your experimental results? Looking forward to your reply.

**Comment by chevalierNoir** [comment]
Hi,

Regarding the inferior performance of AV-HuBERT to the audio-HuBERT in ASR under clean setting, we also noticed such phenomenon in our paper (last paragraph of section 4.5) and we attribute it to the hyperparameter selection based on lip-reading rather than on ASR. For Table 4, we trained an audio-HuBERT for one iteration using the clusters from the AV-HuBERT of the last iteration.

1. In our previous experiments, we noticed setting different modality dropout values in the first 4 iterations does not have an impact as large as the one on the last iteration. We will re-check those values in our original configurations for the first 4 iterations on LRS3.
2. `mask_prob_image` in the config file is the probability of each frame being masked, which is equal to $p\times l$ in the paper. Note the `p` in the paper is the probability of one frame being selected as the start.

**Comment by li563042811** [comment]
Thank you for your reply. I think maybe the reason why the result of AV-HuBERT I trained didn't surpass A-HuBERT in your paper is that I used 8 A-100 for pretrain and finetune and I didn't change batch_size and max_tokens.
In your paper, you used 32 GPUs. So the number of epochs of my training is less than yours.

**Comment by chevalierNoir** [comment]
Hi,

In LRS3 pretraining, the model is trained for 400K steps. The training steps for finetuning depend on the task and are 30K for VSR and AVSR under 30 hours. You can find those numbers in the `optimization.max_update` field of the corresponding configuration files. If you use 8 GPUs, you can simulate the 32-GPU pretraining by appending `+optimization.update_freq=[4]`. This will increase the effective batch size to 4 times larger.

**Comment by li563042811** [comment]
Hi
I pretrained AV-HuBERT in LRS3 by appending optimization.update_freq=[4] still using 8 GPUs. The epoch number changes to 275 from 69.  The 1st iteration costs 102 hours and after finetuning in 30h data the clean av WER comes to 12.56, which is 15.88 in case of optimization.update_freq=[1]. 
I didn't change other hyperparameters.
The pretrain config file is avhubert/conf/pretrain/base_lrs3_iter1.yaml appending optimization.update_freq=[4].
The finetune config file is avhubert/conf/av-finetune/base_noise_pt_noise_ft_30h.yaml appending optimization.update_freq=[4].

I have two questions,
1. In your experiments, is the upgrade in the 1st iteration also work in the last iteration with the same relative improvement？
2. Could you give WER of iteration[1-5] of your AV-HuBERT model pretrained in 433h LRS3 and fine-tuned in 30h data?

**Comment by chevalierNoir** [comment]
Hi,

1. We always used 32 GPUs for pretraining in each iteration and thus didn't know the number of relative improvement by using 32 GPUs compared to using 8 GPUs. But my guess is the gain will hold. The gain from using 32 GPUs is also observed in the original [HuBERT paper](https://arxiv.org/pdf/2106.07447.pdf) (Figure 3).
2. For AVSR/ASR, we only train the model for one iteration (last iteration). For VSR (lip reading), we mostly use CTC for fine-tuning during the first 4 iterations and only use seq2seq finetuning for the last iteration (which outperforms CTC). The CTC numbers for each iteration are in the [paper](https://arxiv.org/pdf/2201.02184.pdf) (first line of Table 2). 

Note for fine-tuning, we always use 8 GPUs and haven't tried 32 GPUs for that.
